### ISSUE TITLE
About the mixed precision calculation of multiple optimizers

### ISSUE THREAD

**Issue by Fuca-Fuda** [Lornatang/SRGAN-PyTorch#44]
Hello!

I noticed that you have recently removed the gradient clipping from the Automatic Mixed Precision, I would really like to ask you why you made this change, is it because the gradient clipping is not suitable for this case?

Thank you very much!

**Comment by Fuca-Fuda** [comment]
In addition, I am very sorry but allow me to point out that since you are using nn.BCEWithLogitsLoss(), real_label is also sigmoid processed, so I think the value of real_label is not 1.0 when calculating the BCE.

Thank you very much!

**Comment by Fuca-Fuda** [comment]
In addition, I am very sorry but allow me to point out that since you are using nn.BCEWithLogitsLoss(), real_label is also sigmoid processed, so I think the value of real_label is not 1.0 when calculating the BCE.

Thank you very much!

**Comment by Lornatang** [comment]
BCEW() is more stable than BCE. BCELoss was used before, and now it is no problem to use BCEW with the modified discriminator architecture.

**Comment by Lornatang** [comment]
What is your good opinion on this? Or if you want to improve, please submit a PR
